pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -211,6 +212,34 @@ func (bdl *BlockDeviceList) Filter(predicateKeys ...string) *BlockDeviceList {
 	return filteredBlockDeviceList
 }
 
+// GetUsableBlockDevices returns custom blockdevice list which can be used
+func (bdl *BlockDeviceList) GetUsableBlockDevices(spcName, namespace string) (*BlockDeviceList, error) {
+	// Initialize filtered block device list
+	if bdl == nil {
+		return nil, errors.Errorf("failed to get usable blockdevices: blockdevicelist is nil")
+	}
+	filteredBlockDeviceList := &BlockDeviceList{
+		BlockDeviceList: &ndm.BlockDeviceList{},
+		errs:            nil,
+	}
+	bdcClient := bdc_v1alpha1.NewKubeClient().WithNamespace(namespace)
+	for _, bdObj := range bdl.Items {
+		if bdObj.Status.ClaimState == ndm.BlockDeviceClaimed {
+			bdcName := bdObj.Spec.ClaimRef.Name
+			bdcObj, err := bdcClient.Get(bdcName, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
```

Comment:
It is better to add a little context to the error here .

Reply:
Updated